myapp/app/models/user.rb

```diff
@@ -5,4 +5,5 @@ class User < ApplicationRecord
 
   has_many :tasks, dependent: :destroy
   validates :username, format: { with: /\A[^\s]*\z/, message: 'should not contain space!' }
+  validates :username, uniqueness: { message: 'duplicated. Please use another username!' }
```

Comment:
test cases would be saver based on validations

Reply:
Added test case in 0b32528